# HG changeset patch
# User E. Joshua Rigler <relgire@gmail.com>
# Date 1251323131 14400
#      Wed Aug 26 17:45:31 2009 -0400
# Node ID 19124db6fc1c1488ffe2d805089075126e61cef4
# Parent  be3ee885739b811943cee3c96f74f0b390f743f3
datestr: set tm.isdst to -1 before calling mktime

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,8 +1,12 @@
+2009-08-26  E. Joshua Rigler  <relgire@gmail.com>
+
+	* time/datestr.m: Set tm.isdst to -1 before calling mktime.
+
 2009-08-25  John W. Eaton  <jwe@octave.org>
 
 	* plot/__gnuplot_open_stream__.m: Save pid in __plot_stream__ property.
 	* plot/gnuplot_drawnow.m: Wait for gnuplot subprocess when printing.
 	From Ben Abbott <bpabbott@mac.com>, Rob Mahurin <rob@utk.edu>, and
 	Dmitri Sergatskov <dasergatskov@gmail.com>.
 
 2009-08-22  David Bateman  <dbateman@free.fr>
diff --git a/scripts/time/datestr.m b/scripts/time/datestr.m
--- a/scripts/time/datestr.m
+++ b/scripts/time/datestr.m
@@ -272,17 +272,19 @@ function retval = datestr (date, f, p)
     tm.year = vi(1) - 1900;
     tm.mon = vi(2) - 1;
     tm.mday = vi(3);
     tm.hour = vi(4);
     tm.min = vi(5);
     sec = vi(6);
     tm.sec = fix (sec);
     tm.usec = fix (rem (sec, 1) * 1e6);
-
+    ## Force mktime to check for DST.
+    tm.isdst = -1
+ 
     str = strftime (df, localtime (mktime (tm)));
 
     if (i == 1)
       retval = str;
     else
       retval = [retval; str];
     endif
 
